templates/web/guides/models/defining.md.erb

```diff
@@ -188,6 +188,70 @@ $dut.nvm.reg1.address        # => 0x30
 $dut.nvm.reg1.data           # => 0xFFFF_FFFF
 ~~~
 
+#### Adding Sub Block Groups
+
+If needing to add several similar sub blocks, a `sub_block_group` can be used.
+
+~~~ruby
+# lib/soc/eagle_m352.rb
+def initialize(options={})
+  sub_block_group :memories
+    sub_block :memory0,        class_name: "MEMORY_128_B954"
+    sub_block :memory1,        class_name: "MEMORY_64_B954"
+    sub_block :memory2,        class_name: "MEMORY_32_B954"
+  end
+end
+
+$dut = SOC::EAGLE_M352.new
+$dut.memory0.class               # MEMORY_128_B954
+$dut.memory1.class               # MEMORY_64_B954
+$dut.memory2.class               # MEMORY_32_B954
+$dut.memories.class              # Array (Array of all 3 memory sub_blocks)
+~~~
+
+In addition, a unique container class can also be used instead of an Array to contain the sub_block objects if desired.
+
+The only requirement is that the method `<<` exist for this container class.
+
+Example of using defining a sub_block_group with a custom container class:
+
+~~~ruby
+# lib/soc/eagle_m352.rb
+def initialize(options={})
+  sub_block_group :memories, class_name: "MEMORIES_B294"
+    sub_block :memory0,        class_name: "MEMORY_128_B954"
+    sub_block :memory1,        class_name: "MEMORY_64_B954"
+    sub_block :memory2,        class_name: "MEMORY_32_B954"
+  end
+end
+
+
+$dut = SOC::EAGLE_M352.new
+$dut.memory0.class               # MEMORY_128_B954
+$dut.memory1.class               # MEMORY_64_B954
+$dut.memory2.class               # MEMORY_32_B954
+$dut.memories.class              # MEMORIES_B954 (custom class containing all 3 memory sub_blocks)
+
+~~~
+
+Here's an example of a custom container class like this:
+
+~~~ruby
+# lib/memories.rb
+class MEMORIES_B294 < ::Array
+
+  def <<(sub_block)   # to meet sub_block_group requirement
```

Comment:
I don't think this example should actually define the << method, by inheriting from `Array` you get that for free and don't need to define it.

I would highlight a bit more what is going in here with respect to `< ::Array`, something like:

The recommended way to create a custom container is to sub-class array and then add additional methods, like this:

~~~ruby
# lib/memories.rb
class MEMORIES_B294 < ::Array
  # Example of a custom method to print the names of all contained memories
  def print_ids
    # Since this class is based on an array, we can call use all array methods like 'each' to iterate
    # on the contained objects
    each { |mem| puts mem.id }
  end
end

sub_block_group :memories, class_name: "MEMORIES_B294"
  sub_block :memory0,        class_name: "MEMORY_128_B954"
  sub_block :memory1,        class_name: "MEMORY_64_B954"
  sub_block :memory2,        class_name: "MEMORY_32_B954"
end

# All conventional array methods are supported since we sub-classed Array
memories.size                   # => 3
memories.first                  # => <memory 0 instance>
memories.map { |mem| mem.id }   # => [:memory0, :memory1, :memory2]

# Plus we have our custom method(s)
memories.print_ids
=>
memory0
memory1
memory2
~~~

Reply:
Cool thanks.  Sure that can be a requirement.  But I think maybe added commented out `def <<(sub_block)` as an example to show in case you do want to do want to include extra stuff upon pushing?   It's what I'm doing in current project.

Reply:
Well in that case then I would make the example do something and also take out the comment "to meet sub_block_group requirement" which is mis-leading in this case.